### PR TITLE
Update rabbitmq doc

### PR DESCRIPTION
**Pull request by 1pretz1**
This updates the main RabbitMQ documentation. On improving this doc I've
moved the `Nomenculture` section from the README in
[govuk_message_queue_consumer] into source/manual/rabbitmq.html.md as i
felt it was more at home here. I've linked to it from the consumer gem
so the information is still accessible there.

The original discussion around the decisions made in this doc can be
found [here][original_pr].

Related PR:
[govuk_message_queue_consumer]

Trello:
https://trello.com/c/0OHZT3cq/458-investigate-rabbitmq-and-improve-the-documentation

[govuk_message_queue_consumer]: https://github.com/alphagov/govuk_message_queue_consumer/pull/67
[original_pr]: https://github.com/alphagov/govuk-developer-docs/pull/2733

### CODE REVIEW

File: source/manual/rabbitmq.html.md [hunk 1]
```diff
@@ -4,147 +4,123 @@ title: Manage RabbitMQ
...
+be [publishing-api](https://github.com/alphagov/publishing-api).
+
+**Exchanges** are AMQP entities where messages are sent. They take a message
+and route it into zero or more queues. The routing algorithm used depends on
+the exchange type and rules called _bindings_.  When a content change is made
+in a publishing application (e.g. Travel Advice Publisher), Publishing API
+[publishes a message][publishing_api_publishes_message] to our main exchange,
+`publishing_documents`.
+
+**Queues** in the AMQP model queues are very similar to queues in other message
```
Comment (benthorner) [inline source/manual/rabbitmq.html.md:37]:
```suggestion
**Queues** are very similar to queues in other message
```

File: source/manual/rabbitmq.html.md [hunk 2]
```diff
@@ -4,147 +4,123 @@ title: Manage RabbitMQ
...
+and task-queueing systems: they store messages that are consumed by
+applications. An example of a queue is `email_alert_service` which is used by
+[email-alert-service][email-alert-service] to forward publishing activity to
+email-alert-api. Queues are [created by consumer
+applications][create_queues].
+
+**Bindings** are rules that exchanges use (among other things) to route
+messages to queues. To instruct an exchange E to route messages to a queue Q, Q
+has to be bound to E. Bindings may have an optional routing key attribute. An
+example of a binding is the `cache_clearing_service-high` queue is bound to the
```
Comment (benthorner) [inline source/manual/rabbitmq.html.md:47]: Minor: would be helpful to have a link for where this binding is defined.

File: source/manual/rabbitmq.html.md [hunk 3]
```diff
@@ -4,147 +4,123 @@ title: Manage RabbitMQ
...
+  `"application/json"`
+* routing_key (string) - matches against bindings to filter messages to certain
+  queues. E.g `"guide.major"`
+* persistant (boolean) - tells RabbitMQ whether to save the message to disk.
+
+Message options are set when a message is published. In our use case, the
+message's payload is the content item in JSON format. The code in the
+publishing-api to publish a message is [here][publish_message_call].
+
+**Consumer applications** - like [email-alert-service][email-alert-service] are
```
Comment (benthorner) [inline source/manual/rabbitmq.html.md:67]: Minor: could delete this first sentence, as already mentioned above.

File: source/manual/rabbitmq.html.md [hunk 4]
```diff
@@ -4,147 +4,123 @@ title: Manage RabbitMQ
...
+
+There is no infinite loop because `max-hops` is set to `1`.
+
+Each cluster has a list of private IP addresses of the other cluster's nodes. The connection
+between Carrenza and AWS travels through the VPN.
+
+Since the nodes in AWS use dynamic IP addresses, they are associated to network interfaces
+with fixed IPs.
+
+## RabbitMQ metrics
```
Comment (benthorner) [inline source/manual/rabbitmq.html.md:100]: Minor: this one link could go with the others at the top of the doc.

File: source/manual/rabbitmq.html.md [hunk 5]
```diff
@@ -4,147 +4,123 @@ title: Manage RabbitMQ
...
+exchange.
+
+While the migration of GOV.UK from Carrenza to AWS is ongoing, we run two RabbitMQ clusters,
+one in each environment.
+
+The `published_documents` exchange is federated in both directions, i.e. the RabbitMQ cluster
+from each provider connects as a client to the exchange in the other provider and forwards
+messages to its own exchange.
+
+There is no infinite loop because `max-hops` is set to `1`.
```
Comment (benthorner) [inline source/manual/rabbitmq.html.md:92]: Minor: all these paragraphs just make the doc look longer. We could combine some of them.

File: source/manual/rabbitmq.html.md [hunk 6]
```diff
@@ -4,147 +4,123 @@ title: Manage RabbitMQ
...
-### Sending heartbeats
-
-Heartbeat messages are sent every minute by cron. Currently, we only
-send heartbeat messages to one exchange: the `published_documents`
-exchange. These heartbeats are sent via a [rake task][heartbeat_rake_task]
-in the `publishing-api` app.
-
-[heartbeat_rake_task]: https://github.com/alphagov/publishing-api/blob/012cb3f1ceb3b18e7059a367cc4030aa0763afb4/lib/tasks/heartbeat_messages.rake
-
-### Checking if the federation is ok
```
Comment (benthorner) [inline source/manual/rabbitmq.html.md:48]: We seem to have lost this detail. Can we preserve it here?

File: source/manual/rabbitmq.html.md [hunk 7]
```diff
@@ -4,147 +4,123 @@ title: Manage RabbitMQ
...
-3. Delete the messages
-
-   > **Note**
-   >
-   > There is a risk that you might delete the wrong message(s). This
-   > is because the contents of the queue may have changed.
-
-   Repeat, but change the "Requeue" option to "No".
-
-## Previewing a message for a document_type
```
Comment (benthorner) [inline source/manual/rabbitmq.html.md:136]: Could we preserve this as part of [the Publishing API doc for RabbitMQ](https://github.com/alphagov/publishing-api/blob/master/docs/rabbitmq.md)?

**Review by benthorner — COMMENTED** [review]
Much easier to review, thanks! Just a couple of more major comments about preserving some of the original doc, which seems to have gotten lost in the rewrite.

File: source/manual/rabbitmq.html.md (hunk 2 above)
Comment (1pretz1) [reply source/manual/rabbitmq.html.md]: Sure, have added a link

File: source/manual/rabbitmq.html.md (hunk 6 above)
Comment (1pretz1) [reply source/manual/rabbitmq.html.md:48]: Yup, have added it back and updated the text a little.

File: source/manual/rabbitmq.html.md (hunk 7 above)
Comment (1pretz1) [reply source/manual/rabbitmq.html.md:136]: Sure, have added it the the Publishing API doc here -> https://github.com/alphagov/publishing-api/pull/1847

**Comment by 1pretz1** [comment]
Thanks for the review @benthorner.

I've addressed all your comments, should be ready for another peak! 👀

File: source/manual/rabbitmq.html.md [hunk 8]
```diff
@@ -4,147 +4,139 @@ title: Manage RabbitMQ
...
+exchange. There is no infinite loop because `max-hops` is set to `1`.
 
-   > **Note**
-   >
-   > There is a risk that you might delete the wrong message(s). This
-   > is because the contents of the queue may have changed.
+### Checking if the federation is ok
 
-   Repeat, but change the "Requeue" option to "No".
+Connect to one of the cluster's nodes through ssh via the jumpbox and run:
```
Comment (benthorner) [inline source/manual/rabbitmq.html.md:101]: `gds govuk connect ssh -e production rabbitmq`

**Review by benthorner — APPROVED** [review]
Nice 🇧🇱

**Pull request merged**